### PR TITLE
refactor: extract text context logic and add tests

### PR DESCRIPTION
**Pull request by sedationh**
- Split getContext into DOM handler and pure text processor
- Extract extractTextContext for reusable text processing
- Add comprehensive test coverage for text context extraction
- Unify error handling in extractTextContext function
- Update comments to English for better code maintainability## Type of Changes

- [ ] ✨ New feature (feat)
- [ ] 🐛 Bug fix (fix)
- [ ] 📝 Documentation change (docs)
- [ ] 💄 UI/style change (style)
- [ ] ♻️ Code refactoring (refactor)
- [ ] ⚡ Performance improvement (perf)
- [ ] ✅ Test related (test)
- [ ] 🔧 Build or dependencies update (build)
- [ ] 🔄 CI/CD related (ci)
- [ ] 🌐 Internationalization (i18n)
- [ ] 🧠 AI model related (ai)
- [ ] 🔄 Revert a previous commit (revert)
- [ ] 📦 Other changes that do not modify src or test files (chore)

## Related Issue

Closes #

## How Has This Been Tested?

- [ ] Added unit tests
- [ ] Verified through manual testing

## Checklist

- [ ] I have tested these changes locally
- [ ] I have updated the documentation accordingly if necessary
- [ ] My code follows the code style of this project
- [ ] My changes do not break existing functionality
- [ ] If my code was generated by AI, I have proofread and improved it as necessary.

## Additional Information

    

---

## Summary by cubic
Refactored text context logic into a pure function and added unit tests to make selection handling in the extension more reliable. Fixed edge cases like empty selection and selection equal to the full text.

- **Refactors**
  - Split getContext into a DOM wrapper and extractTextContext for pure text processing.
  - Unified error handling; updated comments to English.

- **Bug Fixes**
  - Handle empty text/selection and when selection matches the entire text.
  - Improve sentence boundary scanning around the selection.

### CODE REVIEW

File: apps/extension/src/entrypoints/selection.content/utils.ts [hunk 1]
```diff
@@ -1,38 +1,37 @@
...
-  const container = selectionRange.commonAncestorContainer
-  const root = container.nodeType === Node.TEXT_NODE
-    ? container.parentElement
-    : (container as Element | null)
-
-  if (!root) {
-    return { before: '', selection: '', after: '' }
+export function extractTextContext(fullText: string, selection: string) {
+  // Handle edge cases: empty text or empty selection
+  if (selection === '' || fullText === '') {
```
Comment (ananaBMaster) [inline apps/extension/src/entrypoints/selection.content/utils.ts]: 这个可以写入测试

File: apps/extension/src/entrypoints/selection.content/__tests__/utils.test.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    const selection = ' This is a test sentence. Another sentence here.'
+
+    const result = extractTextContext(fullText, selection)
+
+    expect(result).toEqual({
+      before: '',
+      selection: ' This is a test sentence. Another sentence here.',
+      after: '',
+    })
+  })
```
Comment (ananaBMaster) [inline apps/extension/src/entrypoints/selection.content/__tests__/utils.test.ts:29]: 还需要一个测试，测试测我之前提到的那个 bug：

The first sentence. This is a test sentence. Another sentence here.

然后选中 " This is a test sentence."，或者选中 “ This is a test sentence”，重点是要选中 T 前面的 “ ”，并且这个空格前面也有句子

File: apps/extension/src/entrypoints/selection.content/utils.ts [hunk 3]
```diff
@@ -50,6 +49,22 @@ export function getContext(selectionRange: Range) {
...
 
+/**
+ * Get the context sentences for the selected text
+ * TODO: this is a simple version, need to improve
+ */
+export function getContext(selectionRange: Range) {
+  const container = selectionRange.commonAncestorContainer
+  const root = container.nodeType === Node.TEXT_NODE
+    ? container.parentElement
+    : (container as Element | null)
```
Comment (ananaBMaster) [inline apps/extension/src/entrypoints/selection.content/utils.ts]: 不要用 as，直接把条件处理了

File: apps/extension/src/entrypoints/selection.content/utils.ts [hunk 4]
```diff
@@ -1,38 +1,37 @@
...
 
-  // 定义句子边界
-  const boundaries = /[.!?。！？]/g
+  // Handle case where selection equals full text
+  if (index === 0 && selection.length === fullText.length) {
+    return { before: '', selection, after: '' }
+  }
+
+  // Define sentence boundaries
+  const boundaries = /[.!?。！？]/
```
Comment (ananaBMaster) [inline apps/extension/src/entrypoints/selection.content/utils.ts]: 感觉很多情况没考虑到，比如别的语言，或者英语中的 `I'm Mr. Oppo, How are you?`

Comment (ananaBMaster) [reply apps/extension/src/entrypoints/selection.content/utils.ts]: 可以考虑要不要结合一些包比如： https://www.npmjs.com/package/cldr-segmentation，或者 Intl.Segmenter API，https://formatjs.github.io/docs/polyfills/intl-segmenter/

并且添加测试测这些情况。

Comment (sedationh) [reply apps/extension/src/entrypoints/selection.content/utils.ts]: 这个拆 PR 来搞吧

File: apps/extension/src/entrypoints/selection.content/__tests__/utils.test.ts (hunk 2 above)
Comment (sedationh) [reply apps/extension/src/entrypoints/selection.content/__tests__/utils.test.ts:29]: ok，这个之前用这部分以为 bug 修复了，验证了也过了，刚刚又测了一遍发现还是不行，重新写了代码

**Review by ananaBMaster — APPROVED** [review]
